dask/utils.py

```diff
@@ -623,11 +623,12 @@ def dispatch(self, cls):
             # Is a lazy registration function present?
             toplevel, _, _ = cls2.__module__.partition(".")
             try:
-                register = self._lazy.pop(toplevel)
+                register = self._lazy[toplevel]
             except KeyError:
                 pass
             else:
                 register()
+                self._lazy.pop(toplevel, None)
```

Comment:
Wonder if we could actually simplify this a bit

```suggestion
            register = self._lazy.get(toplevel)
            if register is not None:
                 register()
                 self._lazy.pop(toplevel, None)
```

Reply:
lgtm

Reply:
Though I agree this suggestion is equivalent to the current state of this PR, I actually find the `try`/`except` to be a bit more clear as catching a `KeyError` is more explicit in saying "this thing doesn't have this key" than `get` + `is not None` (as there _could_ be a key with a value of `None`). Regardless, this isn't a big deal either way. If this is a blocking comment, then I'm happy to update accordingly.

Reply:
There are other ways to do this if we don't want to rely on `None` (though I don't know how `None` would work given we need to call it like a function later, which would break).

```suggestion
            missing = object()
            register = self._lazy.get(toplevel, missing)
            if register is not missing:
                 register()
                 self._lazy.pop(toplevel, None)
```

Reply:
Sure, but I still find the `try`/`except KeyError:` more clear over defining a separate `missing` object. Again, this is a subjective point and probably not a big deal either way. Is this update needed to get this PR in? If so, happy to update

Reply:
No I don't think it matters. Have just been bitten by unexpected `else` behavior in `try` clauses. This combined with the race condition was what motivated the attempt to simplify. That said, given the existing code works for users, would prefer getting the fix in.